feature/settings/src/main/java/com/google/jetpackcamera/settings/ui/SettingsComponents.kt

```diff
@@ -274,48 +331,68 @@ fun CaptureModeSetting(
+            if (fpsUiState is FpsUiState.Disabled) {
+                testTag(fpsUiState.disabledRationale.first().testTag)
+            }
+        },
         title = stringResource(id = R.string.fps_title),
-        enabled = supportedFps.isNotEmpty(),
+        enabled = fpsUiState is FpsUiState.Enabled,
         leadingIcon = null,
-        description = if (supportedFps.isEmpty()) {
-            stringResource(id = R.string.fps_description_unavailable)
-        } else {
-            when (currentTargetFps) {
+        description = if (fpsUiState is FpsUiState.Enabled) {
+            when (fpsUiState.currentSelection) {
                 FPS_15 -> stringResource(id = R.string.fps_description, FPS_15)
                 FPS_30 -> stringResource(id = R.string.fps_description, FPS_30)
                 FPS_60 -> stringResource(id = R.string.fps_description, FPS_60)
                 else -> stringResource(
                     id = R.string.fps_description_auto
                 )
             }
+        } else {
+            disabledRationaleString((fpsUiState as FpsUiState.Disabled).disabledRationale.first())
         },
         popupContents = {
-            Column(Modifier.selectableGroup()) {
-                Text(
-                    text = stringResource(id = R.string.fps_stabilization_disclaimer),
-                    fontStyle = FontStyle.Italic,
-                    color = MaterialTheme.colorScheme.onPrimaryContainer
-                )
+            if (fpsUiState is FpsUiState.Enabled) {
+                Column(Modifier.selectableGroup()) {
+                    Text(
+                        text = stringResource(id = R.string.fps_stabilization_disclaimer),
+                        fontStyle = FontStyle.Italic,
+                        color = MaterialTheme.colorScheme.onPrimaryContainer
+                    )
 
-                SingleChoiceSelector(
-                    text = stringResource(id = R.string.fps_selector_auto),
-                    selected = currentTargetFps == FPS_AUTO,
-                    onClick = { setTargetFps(FPS_AUTO) }
-                )
-                listOf(FPS_15, FPS_30, FPS_60).forEach { fpsOption ->
                     SingleChoiceSelector(
-                        text = "%d".format(fpsOption),
-                        selected = currentTargetFps == fpsOption,
-                        onClick = { setTargetFps(fpsOption) },
-                        enabled = supportedFps.contains(fpsOption)
+                        text = stringResource(id = R.string.fps_selector_auto),
+                        selected = fpsUiState.currentSelection == FPS_AUTO,
+                        onClick = { setTargetFps(FPS_AUTO) },
+                        enabled = fpsUiState.fpsAutoState is SingleSelectableState.Selectable
                     )
+                    listOf(FPS_15, FPS_30, FPS_60).forEach { fpsOption ->
+                        SingleChoiceSelector(
+                            text = "%d".format(fpsOption),
```

Comment:
Just a question, when using merely simple `%d` like this code, is better than using formatting string?
https://developer.android.com/guide/topics/resources/string-resource#formatting-strings

Reply:
nope, i should be using a formatted string here. thanks for the catch!

Reply:
I've given a review on similar format in here https://github.com/google/jetpack-camera-app/pull/257#discussion_r1728031052
and got an answer.

Then I'd searched it for a while on internet and found this.
https://developer.android.com/reference/java/lang/String.html#format(java.lang.String,%20java.lang.Object[])
> Returns a formatted string using the specified format string and arguments.
>
>The locale always used is the one returned by [Locale.getDefault(Locale.Category)](https://developer.android.com/reference/java/util/Locale#getDefault(java.util.Locale.Category)) with [FORMAT](https://developer.android.com/reference/java/util/Locale.Category#FORMAT) category specified.

String.format had been always get default local for localization.
So It seems to be okay to use `"%d".format()`.

Thank you for answering.